### PR TITLE
(CDAP-6871) Extracted a PrivilegesManager out of Authorizer

### PR DESCRIPTION
**Pull request by bdmogal**
- Authorizer extends PrivilegesManager, so extensions still only have to implement one class
- Added a remote implementation of PrivilegesManager, that is used in StreamRunnable to forward grant and revoke requests to master over HTTP, since Sentry does not support proxy authentication

Jira: [CDAP-6871](https://issues.cask.co/browse/CDAP-6871)
Build: http://builds.cask.co/browse/CDAP-DUT4603

### CODE REVIEW

File: cdap-security/src/main/java/co/cask/cdap/security/authorization/RemotePrivilegesManager.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  @Inject
+  RemotePrivilegesManager(CConfiguration cConf, DiscoveryServiceClient discoveryClient) {
+    super(cConf, discoveryClient, Constants.Service.APP_FABRIC_HTTP);
+  }
+
+  @Override
+  public void grant(EntityId entity, Principal principal, Set<Action> actions) throws Exception {
+    LOG.trace("Making request to grant {} on {} to {}", actions, entity, principal);
+    HttpResponse httpResponse = executeRequest("grant", entity, principal, actions);
+    Preconditions.checkArgument(httpResponse.getResponseCode() == HttpResponseStatus.OK.getCode(),
```
Comment (anwar6953) [inline cdap-security/src/main/java/co/cask/cdap/security/authorization/RemotePrivilegesManager.java]: no need to check that its an OK. `executeRequest` already does that, right

Comment (anwar6953) [inline cdap-security/src/main/java/co/cask/cdap/security/authorization/RemotePrivilegesManager.java]: remainder of the code (besides the other instances of the check) LGTM

**Comment by anwar6953** [comment]
Lgtm

**Comment by bdmogal** [comment]
Build green: http://builds.cask.co/browse/CDAP-DUT4603-4. Tested on a cluster. Merging.

**Pull request merged**